### PR TITLE
To solve lazyloading issue

### PR DESCRIPTION
**Pull request by tomas-knap**
Sometimes it happens that lazy loading does not work properly

### CODE REVIEW

File: backend/pom.xml [hunk 1]
```diff
@@ -79,76 +72,50 @@
...
-                <version>2.3.1</version>
+                <version>2.4</version>
                 <configuration>
                     <archive>
                         <manifest>
                             <addClasspath>true</addClasspath>
                             <mainClass>cz.cuni.mff.xrg.odcs.backend.AppEntry</mainClass>
                             <classpathPrefix>lib/</classpathPrefix>
+                            <classpathLayoutType>custom</classpathLayoutType>
+                            <customClasspathLayout>$${artifact.artifactId}-$${artifact.version}$${dashClassifier?}-$${artifact.extension}/</customClasspathLayout>
```
Comment (tomas-knap) [inline backend/pom.xml]: Why was this needed?

**Comment by tomas-knap** [comment]
@cmader, lib-core/lib modules should not be removed - lib-core contains shared libraries for Core DPUs. Please add it back. Will update the doc to make it more clear. What we can do (in a separate pull request) is to actually merge lib and lib-core modules as there is no purpose for lib module apart from containing lib-core module.

**Comment by tomas-knap** [comment]
@cmader, Regarding the libs for backend, we may put them also inside backend.jar. We left them aside to speed up the development.

**Comment by tomas-knap** [comment]
The build process passed, aspect weaving is working. I added comments to pom.xml to explain that unpacking JARs/removing the signature is a workaround.

File: backend/pom.xml (hunk 1 above)
Comment (cmader) [inline backend/pom.xml]: When extracting dependencies into the lib directory, resource clashes can occur, i.e. if two dependencies pull in identically named files. Therefore each dependency is extracted into a subdirectory of lib. Therefore the classpath needs to be adjusted so that classes can be found in each of these dependency subdirectories.

**Comment by tomas-knap** [comment]
@cmader Regarding putting libs to the backend.jar, will you have time to do that and test it by tomorrow?

File: backend/pom.xml (hunk 1 above)
Comment (tomas-knap) [inline backend/pom.xml]: Thanks for clarification!
